turbopack/crates/turbo-tasks-backend/src/database/lmdb/mod.rs

```diff
@@ -0,0 +1,180 @@
+use std::{borrow::Cow, fs::create_dir_all, path::Path, thread::available_parallelism};
+
+use anyhow::{Context, Result};
+use lmdb::{
+    Database, DatabaseFlags, Environment, EnvironmentFlags, RoTransaction, RwTransaction,
+    Transaction, WriteFlags,
+};
+
+use crate::database::key_value_database::{KeySpace, KeyValueDatabase, WriteBatch};
+
+mod extended_key;
+
+pub struct LmbdKeyValueDatabase {
+    env: Environment,
+    infra_db: Database,
+    data_db: Database,
+    meta_db: Database,
+    forward_task_cache_db: Database,
+    reverse_task_cache_db: Database,
+}
+
+impl LmbdKeyValueDatabase {
+    pub fn new(path: &Path) -> Result<Self> {
+        create_dir_all(path).context("Creating database directory failed")?;
+
+        #[cfg(target_arch = "x86")]
+        const MAP_SIZE: usize = usize::MAX;
+        #[cfg(not(target_arch = "x86"))]
+        const MAP_SIZE: usize = 40 * 1024 * 1024 * 1024;
```

Comment:
```suggestion
        #[cfg(target_pointer_width = "32")]
        const MAP_SIZE: usize = usize::MAX;
        #[cfg(not(target_pointer_width = "32"))]
        const MAP_SIZE: usize = 40 * 1024 * 1024 * 1024;
```

I see you're just moving this around, and I probably missed this in previous reviews, but it's probably better to use `target_pointer_width` here? If only because it's clearer about what's happening?

Also, `usize::MAX` probably isn't okay to use on a 32bit arch? You'd have no memory left to run the program after mmapping the database? You probably need to use something like `usize::MAX / 2`?

Also, I saw we dropped our x86 windows target, so maybe I'm just being silly at this point.